src/main/java/org/mule/service/http/impl/service/server/grizzly/GrizzlyServerManager.java

```diff
@@ -253,8 +285,9 @@ public HttpServer createServerFor(ServerAddress serverAddress, Supplier<Schedule
                              createHttpServerFilter(connectionIdleTimeout, usePersistentConnections, delayedExecutor,
                                                     identifier));
 
-    final ManagedGrizzlyHttpServer grizzlyServer = createManagedServer(schedulerSupplier, serverAddress, identifier);
-    executorProvider.addExecutor(serverAddress, (Supplier<ExecutorService>) grizzlyServer.getDelegate());
+    SchedulerSupplier wrappedSupplier = new SchedulerSupplier(schedulerSupplier, serverAddress, executorProvider);
```

Comment:
extract the common code between this method and the one changed above to a single method